js_src/_dart_helpers.js

```diff
@@ -85,26 +85,17 @@ function _createReactDartComponentClass2(dartInteropStatics, componentStatics) {
       this.dartComponent = dartInteropStatics.initComponent(this, componentStatics);
       this.state = dartInteropStatics.handleGetInitialState(this.dartComponent);
     }
-    // FIXME remove unsafe members when implementing new React 16 lifecycle methods
-    UNSAFE_componentWillMount() {
-      dartInteropStatics.handleComponentWillMount(this.dartComponent, this);
-    }
     componentDidMount() {
       dartInteropStatics.handleComponentDidMount(this.dartComponent);
     }
-    // FIXME remove unsafe members when implementing new React 16 lifecycle methods
-    UNSAFE_componentWillReceiveProps(nextProps) {
-      dartInteropStatics.handleComponentWillReceiveProps(this.dartComponent, nextProps);
-    }
     shouldComponentUpdate(nextProps, nextState) {
       return dartInteropStatics.handleShouldComponentUpdate(this.dartComponent, nextProps, nextState);
     }
-    // FIXME remove unsafe members when implementing new React 16 lifecycle methods
-    UNSAFE_componentWillUpdate(nextProps, nextState) {
-      dartInteropStatics.handleComponentWillUpdate(this.dartComponent, nextProps, nextState);
+    getSnapshotBeforeUpdate(prevProps, prevState) {
+      return dartInteropStatics.handleGetSnapshotBeforeUpdate(this.dartComponent, prevProps, prevState);
```

Comment:
When testing a component without `getSnapshotBeforeUpdate`, I get the following error:
> Warning: _ClockComponent.getSnapshotBeforeUpdate(): A snapshot value (or null) must be returned. You have returned undefined.

I think we need the same undefined check that we have in render a couple lines down, since we can't guarantee in Dart whether its `null` will be converted to `null` or `undefined` in JS (even if you `return null;` explicitly)